### PR TITLE
Aws sdk s3 fix more pr comments

### PR DESCRIPTION
**Pull request by Tenaria**
* Most of these changes are merging in master branch into our branch to fix merge conflicts and keep it up to date
* Otherwise, I removed the indirect dependencies from the s3 go.mod folder as per the comment on the PR
* Also, I removed the startTime from the trace span initialisation since it defaults to time.Now() anyhow
* Fixes emitting/ending the trace in the example.go
* Added godocs to package otels3 in doc.go

I've also asked them a few clarifying questions around these changes so I won't merge them until I get a reply from them about it, but otherwise please take a review - thanks!

### CODE REVIEW

File: instrumentation/github.com/aws/aws-sdk-go/service/go.mod [hunk 1]
```diff
@@ -4,17 +4,9 @@ go 1.15
```
Comment (Tenaria) [inline instrumentation/github.com/aws/aws-sdk-go/service/go.mod:4]: Made changes here

File: instrumentation/github.com/aws/aws-sdk-go/service/go.sum [hunk 2]
```diff
@@ -1,6 +1,5 @@
 github.com/aws/aws-sdk-go v1.35.3 h1:r0puXncSaAfRt7Btml2swUo74Kao+vKhO3VLjwDjK54=
```
Comment (Tenaria) [inline instrumentation/github.com/aws/aws-sdk-go/service/go.sum:1]: made changes here

File: instrumentation/github.com/aws/aws-sdk-go/service/otels3/doc.go [hunk 3]
```diff
@@ -12,4 +12,6 @@
 // See the License for the specific language governing permissions and
```
Comment (Tenaria) [inline instrumentation/github.com/aws/aws-sdk-go/service/otels3/doc.go:12]: made changes here

File: instrumentation/github.com/aws/aws-sdk-go/service/otels3/example/example.go [hunk 4]
```diff
@@ -56,7 +56,7 @@ func main() {
 		context.Background(),
```
Comment (Tenaria) [inline instrumentation/github.com/aws/aws-sdk-go/service/otels3/example/example.go:56]: made changes here

File: instrumentation/github.com/aws/aws-sdk-go/service/otels3/example/go.sum [hunk 5]
```diff
@@ -4,7 +4,6 @@ github.com/aws/aws-sdk-go v1.35.3 h1:r0puXncSaAfRt7Btml2swUo74Kao+vKhO3VLjwDjK54
 github.com/aws/aws-sdk-go v1.35.3/go.mod h1:H7NKnBqNVzoTJpGfLrQkkD+ytBA93eiDYi/+8rV9s48=
```
Comment (Tenaria) [inline instrumentation/github.com/aws/aws-sdk-go/service/otels3/example/go.sum:4]: made changes here

File: instrumentation/github.com/aws/aws-sdk-go/service/otels3/s3.go [hunk 6]
```diff
@@ -63,7 +63,6 @@ func (s *instrumentedS3) PutObjectWithContext(ctx aws.Context, input *s3.PutObje
 		ctx,
```
Comment (Tenaria) [inline instrumentation/github.com/aws/aws-sdk-go/service/otels3/s3.go:63]: made changes here

**Comment by Tenaria** [comment]
I commented on the files I made changes on, the rest were changes due to merging master into the branch

**Review by chenzhihao — APPROVED** [review]

**Pull request merged**